_health-care/_js/_components/date_input.jsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import _ from 'lodash';
+
+class DateInput extends React.Component {
+  constructor() {
+    super();
+    this.state = {month: '', date: '', year: ''};
+    this.handleMonthChange = this.handleMonthChange.bind(this);
+    this.handleDateChange = this.handleDateChange.bind(this);
+    this.handleYearChange = this.handleYearChange.bind(this);
+  }
+
+  componentWillMount() {
+    this.id = _.uniqueId('date-input-');
```

Comment:
It's going to affect our code, but in a good way.

Imagine our model is `window.healtcareApp.state = { first_name: 'albert', insurance: [ {name: 'provider1'} {name: 'provider2'}`.

Each component, on initialization should be pass access to this state Object and on componentWillMount(), should read the initial values from. Then in the onChange handler, it should update the right field.

What this means is that we can actually bring up the component in any panel and have it redisplay the state and accept changes automagically regardless of navigation position.